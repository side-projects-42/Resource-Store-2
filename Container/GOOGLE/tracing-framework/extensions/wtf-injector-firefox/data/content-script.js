/**
 * Copyright 2013 Google, Inc. All Rights Reserved.
 *
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

/**
 * @fileoverview Firefox extension content script.
 *
 * @author benvanik@google.com (Ben Vanik)
 */

(function() {

// console.log utility that doesn't explode when it's not present.
var log = window.console ?
    window.console.log.bind(window.console) : function() {};


/**
 * Injects a <script> tag into the document as early as possible.
 * The hope is that the injected contents run before any of the user code.
 * @param {!Element} script <script> element.
 */
function injectScriptTag(script) {
  // Get some kind of target to put the script in.
  // Only valid documents get body/head, so this is a nice way to ignore bad
  // ones. Try to insert as early as possible.
  var targetElement =
      document.documentElement || document.head || document.body;
  if (targetElement) {
    if (targetElement.firstElementChild) {
      targetElement.insertBefore(script, targetElement.firstElementChild);
    } else {
      targetElement.appendChild(script);
    }
    script.parentNode.removeChild(script);
  }
};


/**
 * Injects a function into the page.
 * @param {!Function} fn Function to inject.
 * @param {Array=} opt_args Arguments array. All must be string serializable.
 */
function injectScriptFunction(fn, opt_args) {
  // Format args as strings that can go in the source.
  var args = opt_args || [];
  for (var n = 0; n < args.length; n++) {
    if (args[n] === undefined) {
      args[n] = 'undefined';
    } else if (args[n] === null) {
      args[n] = 'null';
    } else if (typeof args[n] == 'string') {
      // TODO(benvanik): escape
      args[n] = '"' + args[n] + '"';
    } else if (typeof args[n] == 'object') {
      args[n] = JSON.stringify(args[n]);
    }
  }
  args = args.join(',');

  // TODO(benvanik): escape fn source
  var source = [
    '(' + String(fn) + ')(' + args + ');',
    '// Web Tracing Framework injected function: ' + fn.name,
    '//# sourceURL=x://wtf-injector/' + fn.name
  ].join('\n');

  // Create script tag.
  var script = document.createElement('script');
  script.text = source;

  // Add to page.
  injectScriptTag(script);
};


/**
 * Injects a script file into the page.
 * @param {string} url Script URL.
 * @param {string} rawText Script raw text contents.
 */
function injectScriptFile(url, rawText) {
  var filename = url;
  var lastSlash = url.lastIndexOf('/');
  if (lastSlash != -1) {
    filename = url.substr(lastSlash + 1);
  }

  var source = [
    '(function() {' + rawText + '})();',
    '// Web Tracing Framework injected file: ' + url,
    '//# sourceURL=x://wtf-injector/' + filename
  ].join('\n');

  // Setup script tag with the raw source.
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.text = source;

  // Add to page.
  injectScriptTag(script);
};


/**
 * Sets up two-way communications with the page.
 * This enables proxying to the background page.
 */
function setupCommunications() {
  // Listen for data from the extension.
  self.port.on('extension-event', function(data) {
    // We try to keep the data opaque so we don't waste any time here.
    sendMessage(data);
  });

  // Setup a communication channel with the page via events.
  var localId = String(Number(Date.now() + Math.random()));
  document.defaultView.addEventListener('message', function(e) {
    // The message here is from the wtf.ipc.MessageChannel, and has some header
    // data in it.
    // Note that because we are also sending messages it's possible to get
    // our own messages back, so filter that by localId.
    var packet = e.data;
    if (!packet ||
        !packet['wtf_ipc_connect_token'] ||
        packet['wtf_ipc_sender_token'] == localId) {
      return;
    }

    e.stopPropagation();

    // NOTE: Chrome ports do not support transferrables! Need to convert!

    // Pass through the messages.
    // We trust the Chrome security model and just send the data along.
    var data = packet['data'];
    self.port.emit('page-event', JSON.stringify(data));
  }, true);

  function sendMessage(data) {
    var packet = {
      'wtf_ipc_connect_token': true,
      'wtf_ipc_sender_token': localId,
      'data': data
    };
    document.defaultView.postMessage(packet, '*');
  };
};


/**
 * In-page trace preparation function.
 * This is directly inserted into the page and should run immediately after
 * the library has been loaded.
 *
 * This function is stringified and passed to the page, so expect no closure
 * variables and all arguments must be serializable.
 *
 * @param {!Object.<!Object>} addons A map of URL to addon JSON.
 */
function startTracing(addons) {
  // NOTE: this code is injected by string and cannot access any closure
  //     variables!
  // Register addons.
  for (var url in addons) {
    var name = addons[url]['name'];
    log('WTF Addon Installed: ' + name + ' (' + url + ')');
    wtf.addon.registerAddon(url, addons[url]);
  }

  // Show HUD.
  wtf.hud.prepare();

  // Start recording.
  wtf.trace.start();
};


/**
 * Injects the extension, if needed.
 * This is called on startup in the content-script context.
 */
function main() {
  var wtfOptions = self.options.wtfOptions || {};

  setupCommunications();

  injectScriptFile(
      'wtf_trace_web_js_compiled.js', self.options.wtfScriptContents);
  injectScriptFunction(function(options) {
    wtf.trace.prepare(options);
  }, [
    wtfOptions
  ]);
  injectScriptFunction(startTracing, []);
};


main();


})();
