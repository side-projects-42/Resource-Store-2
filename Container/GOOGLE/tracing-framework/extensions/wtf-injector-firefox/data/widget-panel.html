<!DOCTYPE html>
<html>
  <head>
    <link rel="stylesheet" href="styles.css"></link>
    <style>
      body {
        margin: 8px;
        margin-right: 14px;
      }
    </style>
  </head>
  <body class="k">
    <div class="kButtonBarVertical">
      <a class="kButton buttonShowUi">Show UI</a>
      <br/>
      <hr class="kSeparator">
      <a class="kButton buttonResetSettings">Reset Settings For This URL</a>
      <br/>
      <a class="kButton kButtonAction buttonToggleInjector">Enable For This URL</a>
    </div>
  </body>
</html>
