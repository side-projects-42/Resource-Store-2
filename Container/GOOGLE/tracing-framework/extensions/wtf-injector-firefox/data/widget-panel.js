/**
 * Copyright 2013 Google, Inc. All Rights Reserved.
 *
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

/**
 * @fileoverview Firefox extension widget panel.
 *
 * @author benvanik@google.com (Ben Vanik)
 */


var buttonToggle = document.querySelector('.buttonToggleInjector');
buttonToggle.onclick = function(e) {
  e.preventDefault();
  self.port.emit('perform', 'toggle');
};


var buttonReset = document.querySelector('.buttonResetSettings');
buttonReset.onclick = function(e) {
  e.preventDefault();
  self.port.emit('perform', 'reset');
};


var buttonShowUi = document.querySelector('.buttonShowUi');
buttonShowUi.onclick = function(e) {
  e.preventDefault();
  self.port.emit('perform', 'show_ui');
};


var currentState = null;
self.port.on('show', function(state) {
  currentState = state;
  if (currentState && currentState['enabled']) {
    buttonToggle.innerHTML = 'Disable For This URL';
  } else {
    buttonToggle.innerHTML = 'Enable For This URL';
  }
});

