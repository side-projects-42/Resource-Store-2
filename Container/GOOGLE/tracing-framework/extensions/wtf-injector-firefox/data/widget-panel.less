/**
 * Copyright 2013 Google, Inc. All Rights Reserved.
 *
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

/**
 * @author benvanik@google.com (Ben Vanik)
 */

@import "wtf/ui/styles/reset.less";
@import "wtf/ui/styles/css3.less";
@import "wtf/ui/styles/common.less";
@import "wtf/ui/styles/kennedy.less";

body {
  margin: 10px;
  overflow: hidden;
  white-space: nowrap;
}

.buttonBar {
  min-width: 280px;
}

.buttonToggleInjector {}
.buttonResetSettings {}
.buttonShowUi {}
