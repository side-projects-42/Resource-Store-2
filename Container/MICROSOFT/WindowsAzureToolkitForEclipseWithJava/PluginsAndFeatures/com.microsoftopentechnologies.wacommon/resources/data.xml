<?xml version="1.0" encoding="UTF-8"?>
<data>
<property name="plugin-version" value=""/>
<property name="installation-id" value=""/>
<property name="preference-value" value=""/>
</data>