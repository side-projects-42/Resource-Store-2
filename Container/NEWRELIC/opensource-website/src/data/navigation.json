{
  "navigation": [
    {
      "label": "External Projects",
      "link": "/external-projects",
      "order": 1
    },
    {
      "label": "Highlighted Projects",
      "link": "/collections",
      "order": 2
    },
    {
      "label": "New Relic Projects",
      "link": "/explore-projects",
      "order": 3
    },
    {
      "label": "Standards",
      "link": "/standards",
      "order": 4
    }
  ]
}
