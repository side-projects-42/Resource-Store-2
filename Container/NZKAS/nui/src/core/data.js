/*global nui*/

/**
 * NUI data handling.
 * @class data
 * @static
 * @namespace nui
 */
nui.data = {

    /**
     * Parses a string in the format of "foo:bar; foo:bar"
     * into an object with key-value pairs.
     * @param {String} text
     * @return {Object} An object with key-value pairs.
     * @method parse
     */
    parse: function parseInfo(text){
        var info = {},
            parts = text.split(/\s*;\s*/g),
            i = 0,
            len = parts.length,
            subparts;

        while(i < len){
            subparts = parts[i].split(/\s*:\s*/g);
            info[subparts[0]] = subparts[1];
            i++;
        }

        return info;

    }

};
