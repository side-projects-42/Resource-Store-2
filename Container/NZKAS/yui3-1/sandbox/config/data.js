var configData = {"node":{"submodules":{"node-screen":{"path":"node\/node-screen-min.js","requires":["dom-screen","node-base"],"sizes":{"min":654,"debug":4931,"full":4931},"info":{"name":"node-screen","desc":"None Given"}},"node-style":{"path":"node\/node-style-min.js","requires":["dom-style","node-base"],"sizes":{"min":162,"debug":1199,"full":1199},"info":{"name":"node-style","desc":"None Given"}},"node-base":{"path":"node\/node-base-min.js","requires":["dom-base","selector"],"sizes":{"min":6560,"debug":32088,"full":31936},"info":{"name":"node-base","desc":"None Given"}}}},"queue":{"submodules":{},"path":"queue\/queue-min.js","sizes":{"min":1270,"debug":7340,"full":7340},"info":{"name":"queue","desc":"None Given"}},"anim":{"submodules":{"anim-color":{"path":"anim\/anim-color-min.js","requires":["anim-base"],"sizes":{"min":742,"debug":1320,"full":1320},"info":{"name":"anim-color","desc":"None Given"}},"anim-curve":{"path":"anim\/anim-curve-min.js","requires":["anim-xy"],"sizes":{"min":556,"debug":1795,"full":1795},"info":{"name":"anim-curve","desc":"None Given"}},"anim-easing":{"path":"anim\/anim-easing-min.js","sizes":{"min":2227,"debug":12278,"full":12278},"info":{"name":"anim-easing","desc":"None Given"}},"anim-node-plugin":{},"anim-scroll":{"path":"anim\/anim-scroll-min.js","requires":["anim-base"],"sizes":{"min":385,"debug":824,"full":824},"info":{"name":"anim-scroll","desc":"None Given"}},"anim-xy":{"path":"anim\/anim-xy-min.js","requires":["anim-base","node-screen"],"sizes":{"min":269,"debug":563,"full":563},"info":{"name":"anim-xy","desc":"None Given"}}}},"dom":{"submodules":{"dom-style":{"path":"dom\/dom-style-min.js","requires":["dom-base"],"sizes":{"min":4503,"debug":13345,"full":13249},"info":{"name":"dom-style","desc":"None Given"}},"dom-base":{"path":"dom\/dom-base-min.js","requires":["event"],"sizes":{"min":4731,"debug":20028,"full":19869},"info":{"name":"dom-base","desc":"None Given"}},"dom-screen":{"path":"dom\/dom-screen-min.js","requires":["dom-base","dom-style"],"sizes":{"min":4586,"debug":17185,"full":16820},"info":{"name":"dom-screen","desc":"None Given"}},"selector":{"path":"dom\/selector-min.js","requires":["dom-base"],"sizes":{"min":5872,"debug":19073,"full":18744},"info":{"name":"selector","desc":"None Given"}}}},"attribute":{"submodules":{},"requires":["event"],"path":"attribute\/attribute-min.js","sizes":{"min":3441,"debug":25279,"full":24309},"info":{"name":"attribute","desc":"None Given"}},"dd":{"submodules":{"dd-constrain":{"path":"dd\/dd-constrain-min.js","requires":["dd-drag","dd-proxy"],"sizes":{"min":3421,"debug":13964,"full":13615},"info":{"name":"dd-constrain","desc":"None Given"}},"dd-ddm-base":{"path":"dd\/dd-ddm-base-min.js","requires":["node","base"],"sizes":{"min":2116,"debug":9412,"full":9412},"info":{"name":"dd-ddm-base","desc":"None Given"}},"dd-ddm-drop":{"path":"dd\/dd-ddm-drop-min.js","requires":["dd-ddm"],"sizes":{"min":3390,"debug":13999,"full":13856},"info":{"name":"dd-ddm-drop","desc":"None Given"}},"dd-ddm":{"path":"dd\/dd-ddm-min.js","requires":["dd-ddm-base"],"sizes":{"min":1278,"debug":3660,"full":3660},"info":{"name":"dd-ddm","desc":"None Given"}},"dd-drag":{"path":"dd\/dd-drag-min.js","requires":["dd-ddm-base"],"sizes":{"min":7550,"debug":33132,"full":32012},"info":{"name":"dd-drag","desc":"None Given"}},"dd-drop":{"path":"dd\/dd-drop-min.js","requires":["dd-ddm-drop"],"sizes":{"min":4244,"debug":14149,"full":14149},"info":{"name":"dd-drop","desc":"None Given"}},"dd-proxy":{"path":"dd\/dd-proxy-min.js","requires":["dd-drag"],"sizes":{"min":1558,"debug":5636,"full":5636},"info":{"name":"dd-proxy","desc":"None Given"}}}},"get":{"submodules":{}},"yuitest":{"submodules":{}},"loader":{"submodules":{}},"yui":{"submodules":{},"supersedes":["yui-base","get","loader"],"path":"yui\/yui-min.js","sizes":{"min":20690,"debug":110084,"full":105650},"info":{"name":"yui","desc":"None Given"}},"json":{"submodules":{},"supersedes":["json-parse","json-stringify"],"path":"json\/json-min.js","sizes":{"min":2752,"debug":10584,"full":10584},"info":{"name":"json","desc":"None Given"}},"base":{"submodules":{},"requires":["attribute"],"path":"base\/base-min.js","sizes":{"min":3633,"debug":21929,"full":21773},"info":{"name":"base","desc":"None Given"}},"cookie":{"submodules":{},"path":"cookie\/cookie-min.js","sizes":{"min":3125,"debug":16586,"full":16586},"info":{"name":"cookie","desc":"None Given"}},"dd-plugin":{"submodules":{"dd-drop-plugin":{"path":"dd\/dd-drop-plugin-min.js","requires":["dd-drop"],"sizes":{"min":269,"debug":1116,"full":1116},"info":{"name":"dd-drop-plugin","desc":"None Given"}},"dd-plugin":{"path":"dd\/dd-plugin-min.js","requires":["dd-drag"],"optional":["dd-constrain","dd-proxy"],"sizes":{"min":294,"debug":1129,"full":1129},"info":{"name":"dd-plugin","desc":"None Given"}}}},"profiler":{"submodules":{}},"event":{"submodules":{},"requires":["oop"],"path":"event\/event-min.js","sizes":{"min":16392,"debug":93962,"full":91814},"info":{"name":"event","desc":"None Given"}}};