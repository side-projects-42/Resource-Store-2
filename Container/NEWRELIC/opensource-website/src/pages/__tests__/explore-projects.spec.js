import React from 'react';
import TestRenderer from 'react-test-renderer'; // https://reactjs.org/docs/test-renderer.html

import ExploreProjectsPage from '../explore-projects';
import data from './fixtures/explore-projects';

describe('Explore Projects Page', () => {
  it('Renders correctly', () => {
    const tree = TestRenderer.create(
      <ExploreProjectsPage
        data={data}
        pageContext={{ fileRelativePath: 'src/pages/explore-projects.js' }}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
