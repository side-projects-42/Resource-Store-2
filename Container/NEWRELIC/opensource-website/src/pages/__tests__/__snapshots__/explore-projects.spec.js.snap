// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore Projects Page Renders correctly 1`] = `null`;
